<launch>
    <!-- SN means Service Name -->
    <arg name="device" value="cuda:0"/>
    <arg name="STT_SN" value="STTService"/>
    <arg name="LLMC_SN" value="LLMCService"/>
    <arg name="TF_SN" value="TFService"/>
    <arg name="FER_SN" value="FERService"/>
    <arg name="TTS_SN" value="TTSService"/>

    <rosparam file="$(find ms_pkg)/params.yaml" command="load"/>

    
    <!-- FER<=Facial Expresstion Recogniton -->
    <node pkg="ms_pkg" type="FERServiceServer_script.py" name="FERServiceServerNode" output="screen">
        <param name="service_name" value="$(arg FER_SN)"/>
        <param name="weight_path" value="check_points/FER/learning_180.pt"/>
        <param name="device" value="$(arg device)"/>
    </node>
    
    
    <!-- STT<=Speech To Text -->
    <node pkg="ms_pkg" type="STTServiceServer_script.py" name="STTServiceServerNode" output="screen">
        <param name="service_name" value="$(arg STT_SN)"/>
        <param name="weight_type" value="base"/>
        <param name="device" value="$(arg device)"/>
    </node>


    <!--TF<= Text Filtering  -->
    <node pkg="ms_pkg" type="TFServiceServer_script.py" name="TFServiceServerNode" output="screen">
        <param name="service_name" value="$(arg TF_SN)"/>
        <!-- <param name="recommand_word_set" value="['추천', '제안', '잘나가는', '배고프다']"/> -->
    </node>

    <!--LLMC<= LLM Conversation  -->
    <node pkg="ms_pkg" type="LLMCServiceServer_script.py" name="LLMCServiceServerNode" output="screen">
        <param name="service_name" value="$(arg LLMC_SN)"/>
        <param name="sys_prompt" value=" 너의 목표는 고객의 질문에 잘 대답하고 주문을 받아야해. 너가 할 일은 고객과 대화하기, 고객의 기분에 맞춰서 음료 추천하기, 주문 받기야.메뉴는 마티니,핑크 레이디,섹스온더비치,블루사파이어,허니문,올림픽이 있어. 추가적으로 고객의 고민들 들어주거나 이야기를 들어주면 좋겠어."/>
    </node>

    <!-- TTS<=TextToSpeech -->
    <node pkg="ms_pkg" type="TTSServiceServer_script.py" name="TTSServiceServerNode" output="screen">
        <param name="service_name" value="$(arg TTS_SN)"/>
        <param name="language" value="ko"/>
    </node>



</launch>
